### ISSUE TITLE
Exceptions with large properties

### ISSUE THREAD

**Issue by kingwill27** [RehanSaeed/Serilog.Exceptions#445]
I encountered an issue using Serilog.Exceptions for certain exception types which contain very large properties that exceed the max log size in New Relic (where I ship logs) causing the logs to not be parsed correctly.
* [System.Data.Entity.Core.OptimisticConcurrencyException](https://docs.microsoft.com/en-us/dotnet/api/system.data.entity.core.optimisticconcurrencyexception?view=entity-framework-6.2.0) { StateEntries }
* [AutoMapper.AutoMapperMappingException](https://github.com/AutoMapper/AutoMapper/blob/master/src/AutoMapper/AutoMapperMappingException.cs) { TypeMap }

In the first case, I am logging a custom Exception with an InnerException of OptimisticConcurrencyException, so the JSON-formatted log looks like this:

```json
{
  ...
  "Message.Properties.ExceptionDetail": {
    ...
    "InnerException": {
      ...
      "StateEntries": [...],
      "Type": "System.Data.Entity.Core.OptimisticConcurrencyException"
    },
    "Type": "MyCustomExceptionType"
  }
}
```

In the second case, the AutoMapper exception itself is being logged:

```json
{
  ...
  "Message.Properties.ExceptionDetail": {
    ...
    "TypeMap": {...},
    "Type": "AutoMapper.AutoMapperMappingException"
  }
}
```

I'm happy to make contributions to handle both these exceptions without logging these large properties, but wanted to confirm the best approach before starting.

1. My initial thought was to write custom destructurers (one for each exception) that explicitly ignore the large properties. Is this the best way to handle it?
2. Does the nested nature of the first case present any additional concerns, or can it be handled in the same way as the second case?

Also, happy to break this into 2 separate issues if that is preferable.

**Comment by RehanSaeed** [comment]
Thanks for reporting. 

Regarding the first type `System.Data.Entity.Core.OptimisticConcurrencyException`, am I right in thinking thats the older EF6? If its EF Core, we have an EF package that we should add a custom destructurer to.

We could also add a new package and handle AutoMapper as we have done for Refit. Although, having to add custom destructurers for third party libraries has always left a bad taste in my mouth.

Also FYI, I believe you can use Serilog.Expressions to filter out long properties yourself.

https://github.com/serilog/serilog-expressions

**Comment by RehanSaeed** [comment]
Feel free to reopen if you have more questions or info.

**Comment by RehanSaeed** [comment]
Duplicate of https://github.com/RehanSaeed/Serilog.Exceptions/issues/397#issuecomment-1132647325 where I've added a suggested solution.
